recipes/root/all/conanfile.py

```diff
@@ -0,0 +1,210 @@
+from typing import List
+
+from conans import CMake, ConanFile, tools
+
+
+class PythonOption:
+    OFF = "off"
+    SYSTEM = "system"
+    # in future we may allow the user to specify a version when
+    # libPython is available in Conan Center Index.
+    ALL = [OFF, SYSTEM]
+    DEFAULT = OFF
+
+
+class RootConan(ConanFile):
+    name = "root"
+    version = "v6-22-02"
+    license = "LGPL-2.1-or-later"  # of ROOT itself, the recipe is under MIT license.
+    homepage = "https://root.cern/"
+    # ROOT itself is located at: https://github.com/root-project/root
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "CERN ROOT data analysis framework."
+    topics = ("data-analysis", "physics")
+    settings = ("os", "compiler", "build_type", "arch")
+    options = {
+        # Don't allow static build as it is not supported
+        # see: https://sft.its.cern.ch/jira/browse/ROOT-6446
+        # TODO: shared option should be reinstated when hooks issue is resolved
+        # (see: https://github.com/conan-io/hooks/issues/252)
+        # "shared": [True],
+        "fPIC": [True, False],
+        "python": PythonOption.ALL,
+    }
+    default_options = {
+        # "shared": True,
+        "fPIC": True,
+        "libxml2:shared": True,
+        "sqlite3:shared": True,
+        # default python=off as there is currently no libpython in Conan center
+        "python": PythonOption.OFF,
+    }
+    generators = ("cmake_find_package",)
+    requires = (
+        "opengl/system",
+        "libxml2/2.9.10",
+        "glu/system",
+        "xorg/system",
+        "sqlite3/3.33.0",
+        "libjpeg/9d",
+        "libpng/1.6.37",
+        "libcurl/7.73.0",
+        # "libuuid/1.0.3",
+    )
+
+    @property
+    def _rootsrcdir(self) -> str:
+        version = self.version.replace("v", "")
+        return f"root-{version}"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        # Patch ROOT CMake to use Conan SQLITE
+        tools.replace_in_file(
+            f"{self._rootsrcdir}{os.sep}CMakeLists.txt",
+            "project(ROOT)",
+            """project(ROOT)
+            find_package(SQLite3 REQUIRED)
+            set(SQLITE_INCLUDE_DIR ${SQLITE3_INCLUDE_DIRS})
+            set(SQLITE_LIBRARIES SQLite::SQLite)
+            """,
+        )
+        # Fix execute permissions on scripts
+        scripts = [
+            filename
+            for pattern in (
+                f"**{os.sep}configure",
+                f"**{os.sep}*.sh",
+                f"**{os.sep}*.csh",
+                f"**{os.sep}*.bat",
+            )
+            for filename in glob(pattern, recursive=True)
+        ]
+        for s in scripts:
+            self._make_file_executable(s)
+
+    def _make_file_executable(self, filename):
+        st = os.stat(filename)
+        os.chmod(filename, st.st_mode | stat.S_IEXEC)
+
+    @contextmanager
+    def _configure_cmake(self) -> CMake:
+        cmake = CMake(self)
```

Comment:
On a project this size, I would image it might be worthwhile to cache the CMake object

Reply:
OK. 

To check I understand: the suggestion is to save the CMake object as a member of the ConanFile (or RootConan) class when it is first called to prevent repeat calls to configure?

Reply:
Yes that's exactly it, it's only a suggestion, please do not feel obliged

Reply:
No, this is a sensible optimisation as the CMake configure does take some time.

Implemented in 7f503ea.